lib/OpenQA/Schema/Result/Assets.pm

```diff
@@ -106,6 +106,12 @@ sub remove_from_disk {
         log_info("GRU: skipping removal of $type/$name; it does not exist anyways");
         return undef;
     }
+    if (my $target = readlink $file) {
```

Comment:
The preceding if-block let's me ponder: Will we ever delete dangling symlinks? It doesn't look like it because we call `locate_asset` with `mustexit => 1` which does a `-e` check which requires not only the symlink to exist but also its target. So maybe there's also room for improvement when it comes to deleting dangling symlinks.

Reply:
My newly pushed changes will address that problem.